src/varname.jl

```diff
@@ -26,29 +29,64 @@ julia> @varname x[:, 1][1+1]
 x[:,1][2]
 ```
 """
-struct VarName{sym, T<:Tuple}
+struct VarName{sym,T<:Lens}
     indexing::T
 
-    VarName{sym}(indexing::Tuple=()) where {sym} = new{sym,typeof(indexing)}(indexing)
+    function VarName{sym}(indexing=IdentityLens()) where {sym}
+        # TODO: Should we completely disallow or just `@warn` of limited support?
```

Comment:
What do you think @phipsgabler ?

Reply:
Not sure... I think we should allow dynamic representation for generality, but also have a means of "checked construction".  Kind of like the `Symmetric` etc. wrappers in linear algebra work.  

What about giving that functionality to `concretize`, letting it fail for dynamic lenses?  Then we the results of `concretize` and `@varname` are guaranteed to be conrete (or fail), but you can still use the `VarName` constructor directly?